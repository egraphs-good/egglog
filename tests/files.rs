use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
