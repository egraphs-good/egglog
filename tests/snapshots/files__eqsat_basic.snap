---
source: tests/files.rs
expression: serialized
---
nodes:
  String-1976739436905633066:
    op: "\"x\""
    children: []
    eclass: String-1976739436905633066
    cost: 0
  Var-1976739436905633066:
    op: Var
    children:
      - String-1976739436905633066
    eclass: "1"
    cost: 1
  expr2-0:
    op: expr2
    children: []
    eclass: "4"
    cost: 9223372036854776000
  expr1-0:
    op: expr1
    children: []
    eclass: "4"
    cost: 9223372036854776000
  i64-11743562013128004906:
    op: "2"
    children: []
    eclass: i64-11743562013128004906
    cost: 0
  Num-11743562013128004906:
    op: Num
    children:
      - i64-11743562013128004906
    eclass: "0"
    cost: 1
  i64-17615343019692007359:
    op: "3"
    children: []
    eclass: i64-17615343019692007359
    cost: 0
  Num-17615343019692007359:
    op: Num
    children:
      - i64-17615343019692007359
    eclass: "2"
    cost: 1
  i64-16783941965674463102:
    op: "6"
    children: []
    eclass: i64-16783941965674463102
    cost: 0
  Num-16783941965674463102:
    op: Num
    children:
      - i64-16783941965674463102
    eclass: "10"
    cost: 1
  Mul-17615343019692007359:
    op: Mul
    children:
      - Num-11743562013128004906
      - Add-7784354942592584825
    eclass: "4"
    cost: 1
  Mul-5871781006564002453:
    op: Mul
    children:
      - Num-11743562013128004906
      - Var-1976739436905633066
    eclass: "6"
    cost: 1
  Mul-11743562013128004906:
    op: Mul
    children:
      - Num-11743562013128004906
      - Num-17615343019692007359
    eclass: "10"
    cost: 1
  Add-7659469028595837896:
    op: Add
    children:
      - Var-1976739436905633066
      - Num-17615343019692007359
    eclass: "3"
    cost: 1
  Add-7784354942592584825:
    op: Add
    children:
      - Num-17615343019692007359
      - Var-1976739436905633066
    eclass: "3"
    cost: 1
  Add-9842753449732275747:
    op: Add
    children:
      - Mul-5871781006564002453
      - Mul-11743562013128004906
    eclass: "4"
    cost: 1
  Add-11849178328430774015:
    op: Add
    children:
      - Num-16783941965674463102
      - Mul-5871781006564002453
    eclass: "4"
    cost: 1
root_eclasses: []
class_data:
  "1":
    type: Math
  String-1976739436905633066:
    type: String
  "4":
    type: Math
  "0":
    type: Math
  i64-11743562013128004906:
    type: i64
  "2":
    type: Math
  i64-17615343019692007359:
    type: i64
  "10":
    type: Math
  i64-16783941965674463102:
    type: i64
  "3":
    type: Math
  "6":
    type: Math

